### PR TITLE
MES-2293: Cat B+E Journal Data and Vehicle Details on Start Test

### PR DESCRIPTION
**Pull request by szabikr**
## Description

Copying over business details to journal data (Journal -> TestResult)
Copying over vehicle dimensions to vehicle details (Journal -> TestResult)

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [x] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/modules/tests/journal-data/candidate/candidate.action-creator.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      const candidate: CatBEUniqueTypes.Candidate = {
+        ...booking.candidate,
+        businessAddress: get(booking, 'business.businessAddress'),
+        businessName: get(booking, 'business.businessName'),
+        businessTelephone: get(booking, 'business.telephone'),
+      };
+      return new PopulateCandidateDetailsCatBE(candidate);
+    default:
+      // Should throw exception
+      return new PopulateCandidateDetails(booking.candidate);
```
Comment (TomBillingtonUK) [inline src/modules/tests/journal-data/candidate/candidate.action-creator.ts]: Can we throw an exception here - else we won't get back round to working out this bit

File: src/modules/tests/journal-data/candidate/candidate.actions.ts [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-import { Candidate } from '@dvsa/mes-journal-schema';
+import { Candidate } from '@dvsa/mes-test-schema/categories/common';
+import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
 
 export const POPULATE_CANDIDATE_DETAILS = '[JournalEffects] Populate Candidate Details';
 
 export class PopulateCandidateDetails implements Action {
   readonly type = POPULATE_CANDIDATE_DETAILS;
-  constructor(public payload: Candidate) {}
+  constructor(public payload: Candidate | CatBEUniqueTypes.Candidate) {}
```
Comment (TomBillingtonUK) [inline src/modules/tests/journal-data/candidate/candidate.actions.ts]: Does this need to be typed to CatBEUniqueTypes.Candidate

File: src/modules/tests/journal-data/candidate/candidate.action-creator.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { PopulateCandidateDetailsCatBE } from './cat-be/candidate.cat-be.actions';
+import { Booking } from '@dvsa/mes-journal-schema';
+import { get } from 'lodash';
+
+export const createPopulateCandidateDetailsAction = (testCategory: string, booking: Booking) => {
+  switch (testCategory) {
+    case TestCategory.B:
+      return new PopulateCandidateDetails(booking.candidate);
+    case TestCategory.BE:
+      console.log('wer aresgdsbg here sgsdg');
```
Comment (TomBillingtonUK) [inline src/modules/tests/journal-data/candidate/candidate.action-creator.ts]: console log

File: src/modules/tests/journal-data/candidate/candidate.cat-be.reducer.ts [hunk 4]
```diff
@@ -26,7 +26,8 @@ export function candidateCatBEReducer(
   action: candidateActions.Types,
 ): CatBEUniqueTypes.Candidate {
   switch (action.type) {
-    case candidateActions.POPULATE_CANDIDATE_DETAILS:
+    case candidateActions.POPULATE_CANDIDATE_DETAILS_CAT_BE:
+      console.log('We are here in candidateCatBEReducefr');
```
Comment (TomBillingtonUK) [inline src/modules/tests/journal-data/candidate/candidate.cat-be.reducer.ts]: Console log

**Review by TomBillingtonUK — COMMENTED** [review]
Do we need to link the new candidate reducer into a journal-data reducer?

**Review by ammarhaiderbjss — APPROVED** [review]

**Review by MarkPatt89 — APPROVED** [review]

**Pull request merged**